src/vmspawn/vmspawn-scope.h

```diff
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <stdbool.h>
+
+#include "sd-bus.h"
+
+int start_transient_scope(sd_bus *bus, const char *machine_name, bool allow_pidfd, char **ret_scope);
+int run_command_bound_to_scope(sd_bus *bus, const char *scope, const char *service_name, char **cmdline,
+                               char **cleanup_cmdline, char **extra_properties);
```

Comment:
This function should be introduced in either the commit that makes use of it or in a separate commit.

Reply:
currently it isn't used should I split out or just remove it, it was what I had originally written but it seems like socket/service units are just more useful for vmspawn.

Reply:
@sam-leonard-ct Please remove it if not used. You can keep it around in a private branch if you think you might need it in the future